src/runtime/managedtype.cs

```diff
@@ -102,7 +102,7 @@ internal void FreeGCHandle()
                 if ((flags & TypeFlags.HasClrInstance) != 0)
                 {
                     var gc = TryGetGCHandle(new BorrowedReference(ob));
-                    return (ManagedType)gc?.Target;
+                    return gc?.Target as ManagedType;
```

Comment:
I think this is wrong. If gc handle has `Target`, it must be a `ManagedType` instance here. If it is not, this would violate a strong assumption.